src/logic/calculate.js

```diff
@@ -125,6 +125,20 @@ export default function calculate(obj, buttonName) {
     return { operation: buttonName };
   }
 
+  if (buttonName === "←") { // Backspace functionality
+    if (obj.next) {
+      return { next: obj.next.slice(0, -1) };
+    }
+    return {};
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The backspace functionality seems to be working fine. However, it doesn't handle the case when `obj.next` is an empty string or null. This could potentially lead to unexpected behavior. Consider adding a check for these cases.

```diff
  if (buttonName === "←") { // Backspace functionality
    if (obj.next && obj.next.length > 0) {
      return { next: obj.next.slice(0, -1) };
    }
    return {};
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->